client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/SearchDocumentationIT.java

```diff
@@ -199,6 +207,78 @@ public void onFailure(Exception e) {
         }
     }
 
+    @SuppressWarnings({ "unused", "unchecked" })
+    public void testSearchRequest_Aggregations() throws IOException {
+        RestHighLevelClient client = highLevelClient();
+        {
+            BulkRequest request = new BulkRequest();
+            request.add(new IndexRequest("posts", "doc", "1")
+                    .source(XContentType.JSON, "company", "Elastic", "age", 20));
+            request.add(new IndexRequest("posts", "doc", "2")
+                    .source(XContentType.JSON, "company", "Elastic", "age", 30));
+            request.add(new IndexRequest("posts", "doc", "3")
+                    .source(XContentType.JSON, "company", "Elastic", "age", 40));
+            request.setRefreshPolicy(WriteRequest.RefreshPolicy.WAIT_UNTIL);
+            BulkResponse bulkResponse = client.bulk(request);
+            assertSame(bulkResponse.status(), RestStatus.OK);
+            assertFalse(bulkResponse.hasFailures());
+        }
+        {
+            SearchRequest searchRequest = new SearchRequest();
+            // tag::search-request-aggregations
+            SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+            TermsAggregationBuilder aggregation = AggregationBuilders.terms("by_company").field("company.keyword")
+                    .subAggregation(AggregationBuilders.avg("average_age").field("age"));
+            searchSourceBuilder.aggregation(aggregation);
+            // end::search-request-aggregations
+            searchSourceBuilder.query(QueryBuilders.matchAllQuery());
+            searchRequest.source(searchSourceBuilder);
+            SearchResponse searchResponse = client.search(searchRequest);
+            {
+                // tag::search-request-aggregations-get
+                Aggregations aggregations = searchResponse.getAggregations();
+                Terms byCompanyAggregation = aggregations.get("by_company");
+                Bucket elasticBucket = byCompanyAggregation.getBucketByKey("Elastic");
+                Avg averageAge = elasticBucket.getAggregations().get("average_age");
+                double avg = averageAge.getValue();
+                // end::search-request-aggregations-get
+
+                // tag::search-request-aggregations-get-wrongCast
+                try {
+                    Range range = aggregations.get("by_company"); // <1>
+                } catch (ClassCastException ex) {
+                    // expected
```

Comment:
I wouldn't put asserts in the docs snippets, but I wouldn't have this catch either in a snippet. Is that what we'd expect users to do?